src/components/ha-bar-slider.ts

```diff
@@ -0,0 +1,410 @@
+          this.orientation === "vertical"
+        );
+        this.value = this.steppedValue(this.percentageToValue(percentage));
+        fireEvent(this, "value-changed", { value: this.value });
+      });
+    }
+  }
+
+  private get _tenPercentStep() {
+    return Math.max(this.step, (this.max - this.min) / 10);
+  }
+
+  _handleKeyDown(e: KeyboardEvent) {
+    if (!A11Y_KEY_CODES.has(e.code)) return;
+    e.preventDefault();
+    switch (e.code) {
+      case "ArrowRight":
+      case "ArrowUp":
+        this.value = this.boundedValue((this.value ?? 0) + this.step);
+        break;
+      case "ArrowLeft":
+      case "ArrowDown":
+        this.value = this.boundedValue((this.value ?? 0) - this.step);
+        break;
+      case "PageUp":
+        this.value = this.steppedValue(
+          this.boundedValue((this.value ?? 0) + this._tenPercentStep)
+        );
+        break;
+      case "PageDown":
+        this.value = this.steppedValue(
+          this.boundedValue((this.value ?? 0) - this._tenPercentStep)
+        );
+        break;
+      case "Home":
+        this.value = this.min;
+        break;
+      case "End":
+        this.value = this.max;
+        break;
+    }
+    fireEvent(this, "slider-moved", { value: this.value });
+  }
+
+  _handleKeyUp(e: KeyboardEvent) {
+    if (!A11Y_KEY_CODES.has(e.code)) return;
+    e.preventDefault();
+    fireEvent(this, "value-changed", { value: this.value });
+  }
+
+  destroyListeners() {
+    if (this._mc) {
+      this._mc.destroy();
+      this._mc = undefined;
+    }
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <div
+        id="slider"
+        class=${classMap({
+          slider: true,
+          controlled: this.controlled,
+        })}
+        style=${styleMap({
+          "--value": `${this.valueToPercentage(this.value ?? 0)}`,
+        })}
+        tabindex="0"
+        @keydown=${this._handleKeyDown}
+        @keyup=${this._handleKeyUp}
+        role="slider"
+        aria-valuemin=${this.min}
+        aria-valuemax=${this.max}
+        aria-valuenow=${this.value ?? 0}
+        aria-labelledby=${ifDefined(this.label)}
+      >
+        <div class="slider-track-background"></div>
+        ${this.trackMode === "indicator"
+          ? html`<div
+              class=${classMap({
+                "slider-track-indicator": true,
+                [this.orientation ?? "horizontal"]: true,
+              })}
+            ></div> `
+          : html`<div
+              class=${classMap({
+                "slider-track-bar": true,
+                [this.orientation ?? "horizontal"]: true,
+                [this.trackMode ?? "start"]: true,
+              })}
+            ></div>`}
+      </div>
+    `;
+  }
+
+  static get styles(): CSSResultGroup {
+    return css`
+      :host {
+        display: block;
+        --main-color: rgba(var(--rgb-primary-color), 1);
+        --bg-gradient: none;
+        --bg-color: rgba(var(--rgb-secondary-text-color), 0.2);
```

Comment:
Can we rename this variables so it is clear they are for the bar slider?
Like:
```
bar-slider-track-color
bar-slider-bar-color
etc
```

Reply:
I just renamed it locally before reading your comment 😅